service/Users/users.js

```diff
@@ -1,42 +1,33 @@
-const mongo = require('../../utils/mongo');
-
-// ToDo: Make single instance of db object
-// Remove logging of sensitive information
+const mongo = require("../../utils/mongo");
+const user = require("../../model/user");
 
 class Users {
-	constructor() {
-		this.mongo = mongo;
-	}
-
-	async createUser(userObj) {
-		const mongoClient = await mongo.getClient();
-		const dbo = mongoClient.db('sas');
-		await dbo.collection('users').insertOne(userObj).then(
-			(result) => {
-				console.log(`user created successfully${result}`);
-				return result;
-			},
-		).catch((error) => console.log(error));
-	}
-
-	async getUserByExternalId(provider, id) {
-		console.log('fetching external ID');
-		const mongoClient = await mongo.getClient();
-		const dbo = mongoClient.db('sas');
-		const user = await dbo.collection('users').findOne({ providers: { $elemMatch: { provider, id } } });
-		return user;
-	}
+  async createUser(userObj) {
+    try {
+      let mongoClient = await mongo.getClient();
+      let modeltemp = new user(userObj);
+      let done = await modeltemp.save();
+      return done;
+    } catch (error) {
```

Comment:
why not check this first user exist then only create, why we are trusting on error code and 11000 error code is any standard error code ? .let's make it simple just keep check if doesn't exist then insert why to delay and wait for failure .

Reply:
checking this and then creating will give two network call to the database which is any how not better than making single call and this status code is specific to the mongo DB which is the inherent error of the mongoose which we need not to customize

Reply:
Yep agree we have to do only one network call but question is here we are totally dependent on mongo db if in case your` syntax is wrong `it will treat as duplicate record so to avoid this situation still I am up for two network call since we will have proper visibility and control on `Id` on which we want to find record instead of using default `index` of mongo db.

Reply:
however on using mongoose we won't be having issues like wrong query but yes wrong data could be there which is the issue of code. 
we are trying to do CAS which is the default property of mongo which it do internally we need not to do it externally.